### PR TITLE
bugfix: multi-cluster-status-condition

### LINKED ISSUE #3135: multi type conditions in cluster.status.condition filed
**Describe the Bug**
`k get cluster yy -o yaml`
```
apiVersion: cluster.kubesphere.io/v1alpha1
kind: Cluster
metadata:
  annotations:
    kubesphere.io/creator: admin
    openpitrix.io/runtime: yy
  creationTimestamp: "2020-11-19T14:26:04Z"
  finalizers:
  - finalizer.cluster.kubesphere.io
  generation: 27
  name: yy
  resourceVersion: "10646044"
  selfLink: /apis/cluster.kubesphere.io/v1alpha1/clusters/yy
  uid: 6330f81f-2898-4b4c-afdd-7d483a0a5e0f
spec:
  connection:
    kubeconfig: YXBpVmVyc2lvbjogdjEKY2x1c3RlcnM6Ci0gY2x1c3RlcjoKICAgIGNlcnRpZmljYXRlLWF1dGhvcml0eS1kYXRhOiBMUzB0TFMxQ1JVZEpUaUJEUlZKVVNVWkpRMEZVUlMwdExTMHRDazFKU1VONVJFTkRRV0pEWjBGM1NVSkJaMGxDUVVSQlRrSm5hM0ZvYTJsSE9YY3dRa0ZSYzBaQlJFRldUVkpRUZJUWpock5USlpSR3RWUFFvdExTMHRMVVZPUkNCU1UwRWdVRkpKVmtGVVJTQkxSVmt0TFMwdExRbz0K
    kubernetesAPIEndpoint: https://10.233.25.208:6443
    kubernetesAPIServerPort: 6877
    kubesphereAPIEndpoint: http://10.233.25.208:80
    kubesphereAPIServerPort: 16877
    token: 7e9139f7adb9732dfa2347d8dc0ba459dc1c3f0f5c21ec6394437160e
    type: proxy
  joinFederation: true
status:
  conditions:
  - lastTransitionTime: "2020-11-19T15:26:16Z"
    lastUpdateTime: "2020-11-19T15:26:16Z"
    message: Cluster has been initialized
    reason: Initialized
    status: "True"
    type: Initialized
  - lastTransitionTime: "2020-11-19T15:26:16Z"
    lastUpdateTime: "2020-11-19T15:26:16Z"
    message: Cluster has joined federation control plane successfully
    status: "True"
    type: Federated
  - lastTransitionTime: "2020-11-19T15:26:16Z"
    lastUpdateTime: "2020-11-19T15:26:16Z"
    message: Cluster is not available now
    reason: Unable to establish connection with cluster
    status: "False"
    type: Ready
  - lastTransitionTime: "2020-11-19T15:28:18Z"
    lastUpdateTime: "2020-11-19T15:28:18Z"
    message: Cluster is available now
    reason: Ready
    status: "True"
    type: Ready
```
The ks-console will use `cluster.status.condition` field to confirm whether the cluster is ready, there are two conditions for type Ready, sometimes ks-console will display `waiting for member cluster to join`.
**Versions Used**
KubeSphere: v3.0.0
**How To Reproduce**
1. Add a member cluster with tower agent.
2. rollout restart tower agent
3. `k get cluster yy -o yaml`

**Expected behavior**
Type of `Ready` condition should only appear once, not twice.

### PR DESCRIPTION
**Pull request by yuswift**
Signed-off-by: yuswift <yuswiftli@yunify.com>

**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Avoid ks-console displaying `waiting for member cluster to join`.
**Which issue(s) this PR fixes**:

Fixes #3135

### CODE REVIEW

File: pkg/controller/cluster/cluster_controller.go [hunk 1]
```diff
@@ -797,12 +799,11 @@ func (c *clusterController) updateClusterCondition(cluster *clusterv1alpha1.Clus
 		if cond.Type == condition.Type {
 			if cond.Status == condition.Status {
 				needToUpdate = false
-				continue
-			} else {
-				newConditions = append(newConditions, cond)
+				break
```
Comment (zryfish) [inline pkg/controller/cluster/cluster_controller.go:802]: If we break from here, conditions behind `cond` will be deleted. I think we should refactor this function like this. 

Copy all of the conditions to the `newConditions` except those type equal to the argument condition, then add argument condition.
```go
func (c *clusterController) updateClusterCondition(cluster *clusterv1alpha1.Cluster, condition clusterv1alpha1.ClusterCondition) {
	if cluster.Status.Conditions == nil {
		cluster.Status.Conditions = make([]clusterv1alpha1.ClusterCondition, 0)
	}

	newConditions := make([]clusterv1alpha1.ClusterCondition, 0)
	for _, cond := range cluster.Status.Conditions {
		if cond.Type == condition.Type {
			continue
		} 
		newConditions = append(newConditions, cond)
	}

	newConditions = append(newConditions, condition)
	cluster.Status.Conditions = newConditions
}
```

Comment (yuswift) [reply pkg/controller/cluster/cluster_controller.go:802]: We break here and set `needtoupdate` as `false`, so we will not update the conditions. It seems that your suggestion is more simple, which way should we choose?

Comment (zryfish) [reply pkg/controller/cluster/cluster_controller.go:802]: I prefer the simple one :)

Comment (yuswift) [reply pkg/controller/cluster/cluster_controller.go]: Squashed the commits.

**Comment by fafucoder** [comment]
@zryfish @yuswift  This bugfix will result in a new problem when connection type is proxy

If remove the old condition and add new condition to the cluster, always trigger Initialized this update and never trigger subsequent updates
image
image

**Pull request merged**